lib/src/main/java/org/team100/lib/commands/drivetrain/DriveWithProfile2.java

```diff
@@ -112,22 +113,23 @@ public void execute() {
         Rotation2d currentRotation = m_swerve.getState().pose().getRotation();
         // take the short path
         double measurement = currentRotation.getRadians();
-        Optional<Pose2d> opt = m_fieldRelativeGoal.get();
+        Optional<Pose2d> opt = m_fieldRelativeGoalSupplier.get();
         if (opt.isEmpty()) {
+            if (m_fieldRelativeGoal == null) 
             return;
+        } else {
+            m_fieldRelativeGoal = opt.get();
         }
-        Pose2d fieldRelativeGoal = opt.get();
-
         Rotation2d bearing = new Rotation2d(
-                Math100.getMinDistance(measurement, fieldRelativeGoal.getRotation().getRadians()));
+                Math100.getMinDistance(measurement, m_fieldRelativeGoal.getRotation().getRadians()));
```

Comment:
note the word "bearing" means "angle from point A to point B" but the expression here appears to return simply the angular component of the goal pose itself, which isn't the same thing.  what do you mean this to mean?

Reply:
Fixed